PerimeterXModule/Internals/Validators/PXOriginalTokenValidator.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PerimeterX
+{
+	public class PXOriginalTokenValidator : PXCookieValidator
+	{
+		public PXOriginalTokenValidator(PxModuleConfigurationSection config) : base(config)
+		{
+		}
+
+		public override bool Verify(PxContext context, IPxCookie pxCookie)
+		{
+			if (pxCookie == null)
+			{
+				return false;
+			}
+
+			if (!pxCookie.Deserialize())
+			{
+				Debug.WriteLine("Failed to decode orignal token", PxConstants.LOG_CATEGORY);
+				context.OriginalTokenError = CALL_REASON_DECRYPTION_FAILED;
+				return false;
+			}
+
+			context.DecodedOriginalToken = pxCookie.DecodedCookie;
+			context.OriginalUUID = pxCookie.Uuid;
+			context.Vid = pxCookie.Vid;
+
+			if (!pxCookie.IsSecured(config.CookieKey, getAdditionalSignedFields(context)))
```

Comment:
Since this is already a mobile-related flow, you could replace the call to `getAdditionalSignedFields` with `new string[] { }`. If you really want to keep things generic (i.e. perhaps at some point in the future there will be signed fields relevant to mobile) then leave this call, but in practice at the moment it is not required.

Reply:
doing through the getAdditionalSignedFields would keep it generic on this case

Reply:
👍